java-checks/src/test/files/checks/security/SMTPSSLServerIdentityCheck/JavaMailSession.java

```diff
@@ -0,0 +1,58 @@
+import java.util.Properties;
+import javax.mail.Session;
+
+public class JavaMailSession {
+
+  public void foo() {
+    Properties props = new Properties();
+    props.put("mail.smtp.socketFactory.class", "javax.net.ssl.SSLSocketFactory");
+    Session s = Session.getDefaultInstance(props, new javax.mail.Authenticator() { // Noncompliant [[sc=17;ec=43]] {{Enable server identity validation, set "mail.smtp.ssl.checkserveridentity" to true}}
+      protected PasswordAuthentication getPasswordAuthentication() {
+        return new PasswordAuthentication("username@gmail.com", "password");
+      }
```

Comment:
This piece of code to create an `Authenticator` is duplicated 4 times, but it's not relevant to the test. Can we make the test cases shorter so that we can see easily the differences between them?